src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -580,7 +585,12 @@ def _get_final_device_map(device_map, pipeline_class, passed_class_obj, init_dic
     # Obtain a sorted dictionary for mapping the model-level components
     # to their sizes.
     module_sizes = {
-        module_name: compute_module_sizes(module, dtype=torch_dtype)[""]
+        module_name: compute_module_sizes(
+            module,
+            dtype=torch_dtype.get(module_name, torch_dtype.get("default", torch.float32))
```

Comment:
Maybe I'm missing something, but this is equivalent right?
```suggestion
            dtype=torch_dtype.get(module_name, torch.float32)
```